### PR TITLE
Add global attributes revision / features

### PR DESCRIPTION
**Pull request by mfucci**

### CODE REVIEW

File: src/matter/cluster/Cluster.ts [hunk 1]
```diff
@@ -48,5 +50,47 @@ export const OptionalEvent = <FT extends FieldTemplates>(id: number, priority: E
 export interface Attributes { [key: string]: Attribute<any> }
 export interface Commands { [key: string]: Command<any, any> }
 export interface Events { [key: string]: Event<any> }
-export interface Cluster<C extends Commands, A extends Attributes, E extends Events> { id: number, name: string, commands: C, attributes: A, events: E }
-export const Cluster = <C extends Commands, A extends Attributes, E extends Events>({id, name, attributes = <A>{}, commands = <C>{}, events = <E>{} }: { id: number, name: string, attributes?: A, commands?: C, events?: E } ):Cluster<C, A, E> => ({ id, name, commands, attributes, events });
+
+export type GlobalAttributes<F extends BitTemplates> = {
```
Comment (Apollon77) [inline src/matter/cluster/Cluster.ts:54]: If you want to add referecne use  Core Specs 7.13

File: src/matter/cluster/Cluster.ts [hunk 2]
```diff
@@ -48,5 +50,47 @@ export const OptionalEvent = <FT extends FieldTemplates>(id: number, priority: E
...
+} ):Cluster<F, Merge<A, GlobalAttributes<F>>, C, E> => ({
+    id,
+    name,
+    revision,
+    features,
+    commands,
+    attributes: {
+        ...attributes,
+        clusterRevision: Attribute(0xFFFD, UInt16T),
+        featureMap: Attribute(0xFFFD, BitMapT(features)),
```
Comment (Apollon77) [inline src/matter/cluster/Cluster.ts:93]: 0xFFFC !!

File: src/matter/cluster/OnOffCluster.ts [hunk 3]
```diff
@@ -4,12 +4,16 @@
...
-import { BooleanT } from "../../codec/TlvObjectCodec";
+import { Bit, BooleanT } from "../../codec/TlvObjectCodec";
 import { Attribute, Cluster, Command, NoArgumentsT, NoResponseT } from "./Cluster";
 
 export const OnOffCluster = Cluster({
     id: 0x06,
     name: "On/Off",
+    revision: 1,
+    features: {
+        lightingLevelControl: Bit(0),
```
Comment (Apollon77) [inline src/matter/cluster/OnOffCluster.ts:15]: Whats the value if not defined in "Device.ts"? will it be 0?

**Review by Apollon77 — CHANGES_REQUESTED** [review]

File: src/matter/cluster/Cluster.ts (hunk 2 above)
Comment (mfucci) [reply src/matter/cluster/Cluster.ts:93]: Oups... fixed

File: src/matter/cluster/Cluster.ts (hunk 1 above)
Comment (mfucci) [reply src/matter/cluster/Cluster.ts:54]: Added

File: src/matter/cluster/OnOffCluster.ts (hunk 3 above)
Comment (mfucci) [reply src/matter/cluster/OnOffCluster.ts:15]: Features have to be defined in Device.ts as either true or false.
If this becomes too annoying, I can modify the code to have features optional with false as the default.

Comment (Apollon77) [reply src/matter/cluster/OnOffCluster.ts:15]: All fine but then this is missing or?!

Comment (Apollon77) [reply src/matter/cluster/OnOffCluster.ts:15]: in device.ts in the defined onoff cluster there is nothing set for that bit ... but typescript accepts that ...

**Comment by Apollon77** [comment]
Basically ok, but see last conversatzion

File: src/matter/cluster/OnOffCluster.ts (hunk 3 above)
Comment (mfucci) [reply src/matter/cluster/OnOffCluster.ts:15]: You are correct, I forget one piece... Working late at night is not always the best. Fixed now.

**Review by Apollon77 — APPROVED** [review]

**Comment by mfucci** [comment]
Updated after refactoring the new clusters. Please approve again...

**Review by Apollon77 — APPROVED** [review]
great, thank you ... after merge great base for next clusters :-)

**Pull request merged**